lib/mdl.rb

```diff
@@ -103,24 +104,42 @@ def self.run(argv = ARGV)
               'rule' => id,
               'aliases' => rule.aliases,
               'description' => rule.description,
+              'docs' => rule.docs_url,
             }
-          elsif Config[:show_aliases]
-            puts "#{filename}:#{line}: #{rule.aliases.first || id} " +
-                 rule.description.to_s
           else
-            puts "#{filename}:#{line}: #{id} #{rule.description}"
+            linked_id = linkify(printable_id(rule), rule.docs_url)
+            puts "#{filename}:#{line}: #{linked_id} " + rule.description.to_s
           end
         end
+
+        if !$stdout.tty? && !Config[:json] && !docs_to_print.include?(rule)
```

Comment:
Lets switch the order here and add a comment for future us.

```suggestion
        # If we're not in JSON mode (URLs are in the object), and we cannot
        # make real links (checking if we have a TTY is an OK heuristic for that)
        # than instead of making the output ugly with long URLs, we print them at the
        # end. And of course we only want to print each URL once.
        if !Config[:json] && !$stdout.tty? && !docs_to_print.include?(rule)
```